app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,59 @@
+class CompaniesController < ApplicationController
+  def create
+    code = String.random_alphanumeric(8) 
+    while Company.where(code: code).length != 0
```

Comment:
You are trying to generate a unique code. It's generally a very bad experience to join a company just based on some code.